### PR TITLE
Feat/update signing methods and flows

### PR DESCRIPTION
**Pull request by livingrockrises**
# Description

SmartAccountV2 . signMessage() -> activeValidationModule.signMessage() 
SmartAccountV2 . signUserOp() -> calculate userOpHash -> activeValidationModule.signUserOpHash() 
SmartAccountV2 . signUserOpHash() -> activeValidationModule.signUserOpHash() 
SmartAccountV2 . sendUserOp() ->  SmartAccountV2 . signUserOp() ^ 

other changes,

1) optional argument moduleSignerData can be passed in signUserOp, signUserOpHash and sendUserOp (which will be of type SessionParams)
2) accounts always get the full padded signature from modules (already appended with module address etc)
3) modules are now free of chainId. since userOpHash is calculated by account API

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/account/src/BiconomySmartAccountV2.ts [hunk 1]
```diff
@@ -272,21 +276,31 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
     super.validateUserOp(userOp, requiredFields)
     const userOpHash = await this.getUserOpHash(userOp)
 
-    // Note: we could use signUserOp from validationModule
-    const sig = await this.activeValidationModule.signMessage(arrayify(userOpHash))
+    const sig = await this.activeValidationModule.signUserOpHash(userOpHash, moduleSignerInfo)
```
Comment (tomarsachin2271) [inline packages/account/src/BiconomySmartAccountV2.ts:279]: Didn't understand this change. The decoding of signature where module address and rest of the data is decoded, happens in Smart Account on chain. So this responsibility also should be with BiconomySmartAccount class in the SDK. 
Tomorrow if we upgrade the Smart Account which has new decoding logic, then we would need to change the logic in all validation modules in the SDK as well, if the module address is appended by validation modules.

Comment (livingrockrises) [reply packages/account/src/BiconomySmartAccountV2.ts:279]: do you mean for this specific line or juts the fact that signatureWithModuleAddress should be created in SA?

Comment (tomarsachin2271) [reply packages/account/src/BiconomySmartAccountV2.ts:279]: Yeah i meant in general, for all validation modules.. the module address should be appended in SA class.

Comment (livingrockrises) [reply packages/account/src/BiconomySmartAccountV2.ts:279]: ok will make changes.

Comment (livingrockrises) [reply packages/account/src/BiconomySmartAccountV2.ts:279]: done

**Review by tomarsachin2271 — APPROVED** [review]

**Pull request merged**